lib/explorer/data_frame.ex

```diff
@@ -2552,7 +2552,11 @@ defmodule Explorer.DataFrame do
   defmacro filter(df, query) do
     quote do
       require Explorer.Query
-      Explorer.DataFrame.filter_with(unquote(df), Explorer.Query.query(unquote(query)))
+
+      Explorer.DataFrame.filter_with(
+        unquote(df),
+        Explorer.Query.query(Explorer.Backend.LazyFrame.new(unquote(df)), unquote(query))
```

Comment:
We should not unquote a variable twice, if we do that, it means that if I write this code:

```
DF.new()
|> ...
|> ...
|> DF.filter(...)
```

everything on the left side will be executed twice.

I am thinking we should probably not change the `Explorer.Query` API, should we? What if we change `filter_with` instead to accept either an anonymous function or a lazy frame? And we keep Explorer.Query as it is?

Reply:
Oh that's not good... 😬 Great call out.

I think this PR probably won't go forward as is because of https://github.com/elixir-explorer/explorer/pull/989#issuecomment-2366836563, but if it does I'll address this.

Reply:
[fcdad36](https://github.com/elixir-explorer/explorer/pull/989/commits/fcdad369c481e77d636bae4eb8fde86a99c8e1f7)